### PR TITLE
`dcos task exec` [WIP]

### PR DESCRIPTION
**Pull request by malnick**
This PR adds `dcos task exec` functionality for debugging Mesos Tasks by allowing users to fork processes inside the namespace of containers inside a DC/OS cluster. 

This PR is WIP, and currently serves as the reference implementation while the Mesos API which it interacts gets developed. 

TODO: 
- [ ] Task -> container ID mapping in state.json

### CODE REVIEW

File: cli/dcoscli/data/help/task.txt [hunk 1]
```diff
@@ -4,11 +4,15 @@ Description:
 Usage:
     dcos task --help
     dcos task --info
+		dcos task exec [--interactive --pty] <task_id> <cmd>
```
Comment (tamarrow) [inline cli/dcoscli/data/help/task.txt]: this needs to be in line with the `dcos` above for docopt parsing

File: cli/dcoscli/task/main.py [hunk 2]
```diff
@@ -59,6 +59,11 @@ def _cmds():
             hierarchy=['task'],
             arg_keys=['<task>', '--completed', '--json'],
             function=_task),
+
+        cmds.Command(
+            hierarchy=['task', 'exec'],
+            arg_keys=['<task>', '--interactive', '--pty'],
```
Comment (malnick) [inline cli/dcoscli/task/main.py]: TODO @malnick - Needs cmd to run with cmd arguments.

File: dcos/mesos.py [hunk 3]
```diff
@@ -987,8 +992,35 @@ def __init__(self, task, cmd, interactive=False, pty=False):
...
+            self.interactive
+        )
+        # Initilize nested container using Call
+        call_msg = agent_pb2.Call(
+            agent_pb2.Call.LAUNCH_NESTED_CONTAINER_SESSSION,
+            launch_nested_container_msg)
+
+        pickled_msg = pickle.dumps(call_msg)
+
+        request = requests.Request(
```
Comment (tamarrow) [inline dcos/mesos.py]: we should be using `dcos.http` for requests. Its our wrapper around requests- handles auth, ssl, etc.

File: cli/dcoscli/data/help/task.txt (hunk 1 above)
Comment (malnick) [inline cli/dcoscli/data/help/task.txt]: Thanks! Fixed the first bit, on to the tests.

File: dcos/mesos.py (hunk 3 above)
Comment (malnick) [inline dcos/mesos.py]: Fixed. Are there specific tests that should be added for this as well?

**Comment by jdef** [comment]
maybe a comment in the .proto files indicating which mesos version they were last sync'd to? or at least in the commit history for the file?

**Comment by malnick** [comment]
Realized yesterday there will be a JSON endpoint here, so we may end up not needed the proto files after all...

**Comment by malnick** [comment]
@pyronicide @klueska This implements the mocked out `dcos task exec` and satisfies https://mesosphere.atlassian.net/browse/DCOS-10913

**Comment by malnick** [comment]
Closing in favor of #838

**Pull request closed without merging**